### PR TITLE
Uplink Price Rework

### PR DESCRIPTION
**Pull request by matttheficus**
## What Does This PR Do
**OH GOD MATT'S GONNA CRASH THE UPLINK STOCK MARKET PANIC!**

Changes a LARGE amount of items within the Uplink's cost. This was discussed on [Discord HERE](https://discord.com/channels/145533722026967040/1017233317160038420/1017234792879759480) if you would like some insight into our thoughts behind some of the costs. Items marked `(Non-Nukie)` will only be affected for normal Traitors.
<details>
<summary>Traitor Items</summary>
<br>
All 10mm Special Ammo 2TC <br>
Ambrosia Cruciatus 2 > 1 <br>
Booze Shells 3 > 2 <br>
Camera Flash 2 > 1 <br>
DNA Scrambler 4 > 2 <br>
Dehydrated Carp 2 > 1 <br>
EMP Bomb (Non-Nukie) 10 > 8 <br>
EMP Light 2 > 4 <br>
Energized Axe 10 > 8 <br>
Feral Cat Grenade 4 > 3 <br>
Freedom Implant 5 > 6 <br>
Garrote 10 > 6 <br>
Honk Nade 5 > 3 <br>
Jani Landmine 4 > 2 <br>
Meat Cleaver 10 > 8 <br>
Pizza Bomb 5 > 6 <br>
Plasma Gas Grenades (Non-Nukie) 12 > 10 <br>
Poison Pen 2 > 1 <br>
Power Beacon 10 > 6 <br>
Radio Jammer 5 > 4 <br>
Safety Scissors 5 > 3  <br>
Shotgun Cane 10 > 8 <br>
Stimulants 7 > 8 <br>
Syndicate Bomb (Non-Nukie) 11 > 8 <br>
Thermal Drill 3 > 1 <br>
</details>

<details>
<summary>Nukies Items</summary>
<br>
Big Red Button 3 > 1 <br>
Flamethrower 8 > 1 <br>
Reinforcements 25 > 20 <br>
Viscerators 6 > 5 <br>
</details>

And below are some items we deemed out of scope/had underlying issues with the items themselves. I wont be touching them in this PR but they were noted.

<details>
<summary>Items with Issues</summary>
<br>
AI Comms Key (due to Malf) <br>
Blood Red Magboots (make advanced/baseline in SSU) <br>
Nanocalcium (doesnt fix IB, could be cheaper) <br>
Nukie Combat Medic Kit (Defib is stupid) <br>
VTEC (Make it nukie borg standard) <br>
Viral Injector (Trash item due to Viro) <br>
KPA Pressure Mod (Boring, power is variable)
</details>

## Why It's Good For The Game
Many of the items are rarely, if ever, used. Most of them only see niche use from Surplus crates.

## Testing
Compiled and loaded in.

## Changelog
:cl:
tweak: Tweaked the cost of a large amount of Uplink items.
/:cl:

### CODE REVIEW

**Review by Sirryan2002 — APPROVED** [review]
code looks good

**Comment by matttheficus** [comment]
Forgot from the list.
`Nukie Reinforcement: 25 > 20`

File: code/datums/uplink_item.dm [hunk 1]
```diff
@@ -1048,18 +1048,37 @@ GLOBAL_LIST_INIT(uplink_items, subtypesof(/datum/uplink_item))
...
 	reference = "SBEMP"
 	item = /obj/item/radio/beacon/syndicate/bomb/emp
+	cost = 8
+	excludefrom = list(UPLINK_TYPE_NUCLEAR, UPLINK_TYPE_SST)
+	surplus = 0
+	cant_discount = TRUE
+
+/datum/uplink_item/explosives/emp_bomb/nuke
+	reference = "NSBEMP"
+	item = /obj/item/radio/beacon/syndicate/bomb/emp
```
Comment (hal9000PR) [inline code/datums/uplink_item.dm:1079]: dont need to specify the item as it is the same as the subtype.

File: code/datums/uplink_item.dm [hunk 2]
```diff
@@ -1048,18 +1048,37 @@ GLOBAL_LIST_INIT(uplink_items, subtypesof(/datum/uplink_item))
...
-	cost = 11
+	cost = 8
+	excludefrom = list(UPLINK_TYPE_NUCLEAR, UPLINK_TYPE_SST)
 	surplus = 0
 	cant_discount = TRUE
 	hijack_only = TRUE
 
+/datum/uplink_item/explosives/syndicate_bomb/nuke
+	reference = "NSB"
+	item = /obj/item/radio/beacon/syndicate/bomb
```
Comment (hal9000PR) [inline code/datums/uplink_item.dm:1059]: same here

File: code/datums/uplink_item.dm [hunk 3]
```diff
@@ -1124,7 +1143,17 @@ GLOBAL_LIST_INIT(uplink_items, subtypesof(/datum/uplink_item))
...
 	item = /obj/item/storage/box/syndie_kit/atmosfiregrenades
 	hijack_only = TRUE
+	cost = 10
+	excludefrom = list(UPLINK_TYPE_NUCLEAR, UPLINK_TYPE_SST)
+	surplus = 0
+	cant_discount = TRUE
+
+/datum/uplink_item/explosives/atmosfiregrenades/nuke
+	reference = "NAPG"
+	item = /obj/item/storage/box/syndie_kit/atmosfiregrenades
```
Comment (hal9000PR) [inline code/datums/uplink_item.dm:1153]: same here

**Review by hal9000PR — DISMISSED** [review]

**Review by SteelSlayer — APPROVED** [review]

**Pull request merged**